show/bgp_common.py

```diff
@@ -0,0 +1,420 @@
+                        del_cnt += 1
+                    else:
+                        new_nhop_l.append(copy.deepcopy(nh))
+                else:
+                    new_nhop_l.append(copy.deepcopy(nh))
+            # use the new filtered nhop list if it is not empty. if empty nexthop , this route is filtered out completely
+            if len(new_nhop_l) > 0:
+                new_info['nexthops'] = copy.deepcopy(new_nhop_l)
+                new_info_cnt += 1
+                # in case there are any nexthop that were deleted, we will need to adjust the nexhopt counts as well
+                if del_cnt > 0:
+                    internalNextHopNum = new_info['internalNextHopNum'] - del_cnt
+                    new_info['internalNextHopNum'] = internalNextHopNum
+                    internalNextHopActiveNum = new_info['internalNextHopActiveNum'] - del_cnt
+                    new_info['internalNextHopActiveNum'] = internalNextHopActiveNum
+                new_info_l.append(copy.deepcopy(new_info))
+        if new_info_cnt:
+            if asic_cnt > 1:
+                if filter_back_end:
+                    merge_to_combined_route(combined_route, route, new_info_l)
+                else:
+                    new_route[route] = copy.deepcopy(new_info_l)
+            else:
+                new_route[route] = copy.deepcopy(new_info_l)
+    if new_route:
+        if print_ns_str:
+            combined_route['{}'.format(ns_str)] = copy.deepcopy(new_route)
+        else:
+            combined_route.update(copy.deepcopy(new_route))
+
+def print_show_ip_route_hdr():
+    # This prints out the show ip route header based on FRR 7.2 version.
+    # Please note that if we moved to future versions, we may heva to make changes to this
+    print("Codes: K - kernel route, C - connected, S - static, R - RIP,")
+    print("       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,")
+    print("       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,")
+    print("       F - PBR, f - OpenFabric,")
+    print("       > - selected route, * - FIB route, q - queued route, r - rejected route\n")
+
+def show_routes(args, namespace, display, verbose, ipver):
+    import utilities_common.bgp_util as bgp_util
+    """Show IPv4/IPV6 routing table"""
+    if display is None:
+        if multi_asic.is_multi_asic():
+            display = constants.DISPLAY_EXTERNAL
+    else:
+        if multi_asic.is_multi_asic():
+            if display not in multi_asic_util.multi_asic_display_choices():
+                print "dislay option '{}' is not a valid option.".format(display)
+                return
+        else:
+            print "dislay option '{}' is not a valid option.".format(display)
+            return
+    device = multi_asic_util.MultiAsic(display, namespace)
+    arg_strg = ""
+    found_json = 0
+    ns_l = []
+    print_ns_str = False
+    filter_back_end = False
+    filter_by_ip = False
+    asic_cnt = 0
+    if multi_asic.is_multi_asic():
```

Comment:
`device.get_ns_list_based_on_options()`  used below handles both single and multi asic devices. I don't think we need this if condition

Reply:
@arlakshm Done!